metaserver/client.go

```diff
@@ -56,7 +188,702 @@ func (client *Client) RegisterProvider(info *core.ProviderInfo) error {
 	return nil
 }
 
-func (client *Client) RegisterRenter(info *core.RenterInfo) error {
-	// TODO: implement
+func (client *Client) GetRenter(renterID string) (*core.RenterInfo, error) {
+	if client.token == "" {
+		return nil, errors.New("must authorize before calling this method")
+	}
+
+	url := fmt.Sprintf("http://%s/renters/%s", client.addr, renterID)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	token := fmt.Sprintf("Bearer %s", client.token)
+	req.Header.Add("Authorization", token)
+
+	resp, err := client.client.Do(req)
+	if resp.StatusCode != http.StatusOK {
+		var respMsg postRenterResp
+		err = json.NewDecoder(resp.Body).Decode(&respMsg)
+		if err != nil {
+			return nil, err
+		}
+		return nil, errors.New(respMsg.Error)
+	}
+
+	var renter core.RenterInfo
+	err = json.NewDecoder(resp.Body).Decode(&renter)
+	if err != nil {
+		return nil, err
+	}
+	return &renter, nil
+}
+
+func (client *Client) UpdateRenter(renter *core.RenterInfo) error {
+	if client.token == "" {
+		return errors.New("must authorize before calling this method")
+	}
+
+	url := fmt.Sprintf("http://%s/renters/%s", client.addr, renter.ID)
+
+	b, err := json.Marshal(renter)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("PUT", url, bytes.NewReader(b))
+	if err != nil {
+		return err
+	}
+
+	token := fmt.Sprintf("Bearer %s", client.token)
+	req.Header.Add("Authorization", token)
+
+	resp, err := client.client.Do(req)
+	if resp.StatusCode != http.StatusOK {
+		var respMsg postRenterResp
+		err = json.NewDecoder(resp.Body).Decode(&respMsg)
+		if err != nil {
+			return err
+		}
+		return errors.New(respMsg.Error)
+	}
+
+	return nil
+}
+
+func (client *Client) DeleteRenter(renterID string) error {
+	if client.token == "" {
+		return errors.New("must authorize before calling this method")
+	}
+
+	url := fmt.Sprintf("http://%s/renters/%s", client.addr, renterID)
+
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return err
+	}
+
+	token := fmt.Sprintf("Bearer %s", client.token)
+	req.Header.Add("Authorization", token)
+
+	resp, err := client.client.Do(req)
+	if resp.StatusCode != http.StatusOK {
+		var respMsg postRenterResp
+		err = json.NewDecoder(resp.Body).Decode(&respMsg)
+		if err != nil {
+			return err
+		}
+		return errors.New(respMsg.Error)
+	}
+
+	return nil
+}
+
+func (client *Client) PostFile(renterID string, file core.File) error {
```

Comment:
Recommendation: Take file as pointer here and below to avoid copies.

Reply:
Agree.